server/odc-service/src/main/java/com/oceanbase/odc/service/flow/processor/CreateFlowInstanceProcessAspect.java

```diff
@@ -160,4 +172,41 @@ private void adaptCreateFlowInstanceReq(CreateFlowInstanceReq req) {
             config.setDatabaseId(req.getDatabaseId());
         }
     }
+
+    private void adaptCreateFlowInstanceReqForMultipleDatabase(CreateFlowInstanceReq req) {
+        MultipleDatabaseChangeParameters parameters = (MultipleDatabaseChangeParameters) req.getParameters();
+        List<Long> ids = parameters.getOrderedDatabaseIds().stream().flatMap(List::stream).collect(Collectors.toList());
+        // Limit the number of multi-databases change
+        if (ids.size() <= DatabaseChangeProperties.MIN_DATABASE_COUNT || ids.size() > DatabaseChangeProperties.MAX_DATABASE_COUNT) {
+            throw new BadArgumentException(ErrorCodes.IllegalArgument,
+                    "The number of databases must be greater than "+DatabaseChangeProperties.MIN_DATABASE_COUNT+" and not more than "+DatabaseChangeProperties.MAX_DATABASE_COUNT+".");
+        }
+        if (new HashSet<Long>(ids).size() != ids.size()) {
+            throw new BadArgumentException(ErrorCodes.IllegalArgument,
+                    "Database cannot be duplicated.");
+        }
+        List<Database> databases = databaseService.detailForMultipleDatabase(ids);
```

Comment:
This logic is required. It only needs to be queried once, and then it will be used to verify the project, database and data source. The value is set to null during persistence